core/src/main/java/org/elasticsearch/index/CompositeIndexEventListener.java

```diff
@@ -66,182 +64,197 @@ public void shardRoutingChanged(IndexShard indexShard, @Nullable ShardRouting ol
         }
     }
 
-    public void beforeIndexAddedToCluster(Index index, @IndexSettings Settings indexSettings) {
-        for (Listener listener : listeners) {
+    @Override
+    public void afterIndexShardCreated(IndexShard indexShard) {
+        for (IndexEventListener listener : listeners) {
             try {
-                listener.beforeIndexAddedToCluster(index, indexSettings);
+                listener.afterIndexShardCreated(indexShard);
             } catch (Throwable t) {
-                logger.warn("[{}] failed to invoke before index added to cluster callback", t, index.name());
+                logger.warn("{} failed to invoke after shard created callback", t, indexShard.shardId());
```

Comment:
Since we now used a logger with an index prefix , this will result in double index name logging `[index][index][0]`